mlflow/tracing/provider.py

```diff
@@ -164,6 +189,10 @@ def set_destination(destination: TraceDestination):
     A destination specified by this function will take precedence over
     other configurations, such as tracking URI, OTLP environment variables.
 
+    If you set environment variable 'MLFLOW_ENABLE_THREAD_LOCAL_TRACING_DESTINATION' to 'true',
+    then the destination setting only takes effect in current thread,
+    otherwise the destination is set globally.
```

Comment:
```suggestion
    By default, the specified destination is applied globally. To set different destinations
    per thread in multi-threaded application, set the environment variable
    'MLFLOW_ENABLE_THREAD_LOCAL_TRACING_DESTINATION' to 'true',
```

nit: Let's mention which behavior is the default.